cmd/buildah/config.go

```diff
@@ -145,14 +146,22 @@ func updateConfig(builder *buildah.Builder, c *cli.Context) {
 		}
 	}
 	if c.IsSet("entrypoint") {
-		entrypointSpec := make([]string, 3)
-		entrypointSpec[0] = "/bin/sh"
-		entrypointSpec[1] = "-c"
-		entrypointSpec[2] = c.String("entrypoint")
-		if len(strings.TrimSpace(c.String("entrypoint"))) == 0 {
-			builder.SetEntrypoint(nil)
+		var entrypointJSON []string
+		err := json.Unmarshal([]byte(c.String("entrypoint")), &entrypointJSON)
+
+		if err != nil {
+			// it wasn't a valid json array, fall back to string
+			entrypointSpec := make([]string, 3)
+			entrypointSpec[0] = "/bin/sh"
+			entrypointSpec[1] = "-c"
+			entrypointSpec[2] = c.String("entrypoint")
+			if len(strings.TrimSpace(c.String("entrypoint"))) == 0 {
+				builder.SetEntrypoint(nil)
+			} else {
+				builder.SetEntrypoint(entrypointSpec)
+			}
 		} else {
-			builder.SetEntrypoint(entrypointSpec)
+			builder.SetEntrypoint(entrypointJSON)
```

Comment:
Based on the example in the Docker doc that @pixdrift provided in an earlier reply (https://docs.docker.com/engine/reference/builder/#exec-form-entrypoint-example), should we do 'builder.SetCmd(nil)' if we have an array?  I think it still holds for the string case, but I'm not sure about the array case here.

Reply:
I had difficulty determining what the blanking behaviour was replicating in Docker/bud?